.github/workflows/health_checks.yml

```diff
@@ -21,30 +21,43 @@ jobs:
         # creates file system links that include drive letter.
         # Changing between standard and custom workers requires full install cache invalidation
         os: [ubuntu-latest, macos-14, windows-latest]
+        node: [lts/hydrogen, lts/iron]
     runs-on: ${{ matrix.os }}
     timeout-minutes: 10
     steps:
       - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # version 3.6.0
       - uses: ./.github/actions/setup_node
+        with:
+          node-version: ${{ matrix.node }}
       - uses: ./.github/actions/install_with_cache
   build:
+    strategy:
+      matrix:
+        node: [lts/hydrogen, lts/iron]
     runs-on: ubuntu-latest
     needs:
       - install
     steps:
       - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # version 3.6.0
       - uses: ./.github/actions/setup_node
+        with:
+          node-version: ${{ matrix.node }}
+      - uses: ./.github/actions/install_with_cache
       - uses: ./.github/actions/build_with_cache
   test_with_coverage:
     needs:
       - build
     strategy:
       matrix:
         os: [ubuntu-latest, macos-14, windows-latest]
+        node: [lts/hydrogen, lts/iron]
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # version 3.6.0
       - uses: ./.github/actions/setup_node
+        with:
+          node-version: ${{ matrix.node }}
+      - uses: ./.github/actions/install_with_cache
```

Comment:
restore_build_cache already restores the install cache

Reply:
Is there a reason we need both install_with_cache and restore_build_cache now?

Reply:
The tests were failing earlier for missing node_modules, but it looks like its not needed anymore. Removed install_with_cache here.